src/qibolab/instruments/qm/controller.py

```diff
@@ -406,6 +414,11 @@ def preprocess_sweeps(
         for sweeper in find_sweepers(sweepers, Parameter.frequency):
             channels = [(id, self.channels[id]) for id in sweeper.channels]
             find_lo_frequencies(args, channels, configs, sweeper.values)
+            for id, channel in channels:
+                acquisition_id = self._acquisition_id(id)
+                args.parameters[id].element = (
+                    id if acquisition_id is None else acquisition_id
+                )
```

Comment:
The other option is to move this during the channels process, and build a `probe -> acquisition` hash-map from the `AcquireChannel.probe` attribute.

Reply:
Done in bcf9c9df7d732bf9e3cbde73c68d5159c0ea257a. Should be more efficient than searching the channels list multiple times but most likely not that relevant. Thanks.